helper/users/dynamic/pool.go

```diff
@@ -81,6 +94,20 @@ func New(opts *PoolConfig) Pool {
 	}
 }
 
+// noopPool is an implementation of Pool that does not allow acquiring ugids
+type noopPool struct{}
+
+func (*noopPool) Restore(UGID) {}
+func (*noopPool) Acquire() (UGID, error) {
+	return 0, errors.New("dynamic workload users disabled")
```

Comment:
Is it always safe to return an error here? If the `UGID` returned doesn't really matter if the pool is disabled, then it may be safer to just `return 0, nil`.

I once had a noop struct that returned an error because I thought it would never happen, but then it did and broke stuff, so I'm always worried about noops that actually do something somewhere else 😅

Reply:
So the case where this error would be returned is if the user has `exec2` and a task using it, but has their client configured with `users {dynamix_user{min,max} = -1}` (disabling the anonymous workload users functionality). The point is that in this scenario they would be requiring their job submitters to set an actual value for the `task.user` field. If they don't they will get this error.

Perhaps we should set a fingerprint value on whether dynamic workload users are enabled on the client. Although I don't know how we would set a constraint - we would need to know whether the task driver makes use of dynamic workload users at job submission time, and that isn't something we know (sort of, we do know for exec2 because it's ours...)